src/Features/Blockcore.Features.Wallet/Database/WalletStore.cs

```diff
@@ -267,129 +269,389 @@ public IEnumerable<WalletHistoryData> GetAccountHistory(int accountIndex, bool e
             }
+
             return walletBalanceResult;
         }
 
         public WalletBalanceResult GetBalanceForAccount(int accountIndex, bool excludeColdStake)
         {
-            var transactions = this.trxCol
-              .Find(Query.And(Query.EQ("AccountIndex", new BsonValue(accountIndex)), Query.EQ("SpendingDetails", BsonValue.Null)))
-              .Where(x => excludeColdStake && this.network.Consensus.IsProofOfStake ? (x.IsColdCoinStake != true) : true)
-              .GroupBy(x => x.BlockHeight != null)
-              .Select(o =>
-                new
-                {
-                    Confirmed = o.Key,
-                    Amount = o.Sum(x => x.Amount)
-                }).ToList();
+            string excludeColdStakeSql = excludeColdStake && this.network.Consensus.IsProofOfStake ? "AND (IsColdCoinStake = false OR IsColdCoinStake is null) " : string.Empty;
+
+            var sql = "SELECT " +
+                      "BlockHeight as Confirmed," +
+                      "SUM(Amount) as Total " +
+                      "FROM TransactionData " +
+                      $"WHERE SpendingDetailsTransactionId is null AND AccountIndex = {accountIndex} " +
+                      $"{excludeColdStakeSql}" +
+                      $"GROUP BY BlockHeight is not null";
+
+            var result = this.sqliteConnection.Query(sql);
 
             var walletBalanceResult = new WalletBalanceResult();
 
-            foreach (var transaction in transactions)
+            foreach (dynamic item in result)
             {
-                if (transaction.Confirmed == false)
-                    walletBalanceResult.AmountUnconfirmed = transaction.Amount;
-                else
-                    walletBalanceResult.AmountConfirmed = transaction.Amount;
+                if (item.Confirmed == null) walletBalanceResult.AmountUnconfirmed = (long)item.Total;
+                else walletBalanceResult.AmountConfirmed = (long)item.Total;
             }
+
             return walletBalanceResult;
         }
 
         public TransactionOutputData GetForOutput(OutPoint outPoint)
         {
-            var trx = this.trxCol.FindById(outPoint.ToString());
-            return trx;
+            var trx = this.sqliteConnection.QueryFirstOrDefault<TransactionData>(
+                "select * from TransactionData where OutPoint = @outPoint", new { outPoint });
+
+            if (trx == null)
+            {
+                return null;
+            }
+
+            TransactionOutputData ret = this.Convert(trx);
+
+            return ret;
         }
 
         public bool Remove(OutPoint outPoint)
         {
-            return this.trxCol.Delete(outPoint.ToString());
+            var ret = this.sqliteConnection.ExecuteScalar<int>("delete from TransactionData where OutPoint = @outPoint", new { outPoint });
+
+            return ret > 0;
+        }
+
+        private void CreateDatabase()
```

Comment:
Please prepare an "UpdateDatabase" example and explanation on how to make future modifications to the SQL database. Also the version should be written either to a state file on disk, or to a table in the database. Then the version can be queried to know how to perform upgrades, etc.

Reply:
something in the sort of 
> check version if not X then run this script to upgrade.